### PR TITLE
libpanel: init at 1.0.alpha1

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libpanel/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = pname;
+    rev = version;
+    hash = "sha256-jN+NIJxvrAtuH8DN/pdWEmy+/BfXR7Cw2X/MBbLlT/s=";
+  };
+
+  nativeBuildInputs = [
+    # cmake
```
Comment (azahi) [inline pkgs/development/libraries/libpanel/default.nix:31]:
```suggestion
```

File: pkgs/development/libraries/libpanel/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
```
Comment (azahi) [inline pkgs/development/libraries/libpanel/default.nix:4]:
```suggestion
```

File: pkgs/development/libraries/libpanel/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    owner = "GNOME";
+    repo = pname;
+    rev = version;
+    hash = "sha256-jN+NIJxvrAtuH8DN/pdWEmy+/BfXR7Cw2X/MBbLlT/s=";
+  };
+
+  nativeBuildInputs = [
+    # cmake
+    pkg-config
+    glib
```
Comment (azahi) [inline pkgs/development/libraries/libpanel/default.nix:33]: This probably should be in `buildInputs`.

File: pkgs/development/libraries/libpanel/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    meson
+    ninja
+    vala
```
Comment (azahi) [inline pkgs/development/libraries/libpanel/default.nix:36]: This probably should be in `buildInputs`.

File: pkgs/development/libraries/libpanel/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+, vala
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libpanel";
+  version = "1.0.alpha1";
+
+  # outputs = [ "out" "dev" "devdoc" ];
+  # outputBin = "devdoc"; # demo app
+
```
Comment (azahi) [inline pkgs/development/libraries/libpanel/default.nix:21]:
```suggestion
```

**Review by azahi — CHANGES_REQUESTED** [review]

**Comment by jtojnar** [comment]
We have this in https://github.com/NixOS/nixpkgs/pull/182618, including devdocs.

**Comment by magnetophon** [comment]
Closing for #182618.

**Pull request closed without merging**